offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/offergeneration/OfferTemplate.groovy

```diff
@@ -0,0 +1,370 @@
+        document.getElementById("offer-identifier").text(offer.getOfferIdentifier())
+        document.getElementById("offer-expiry-date").text(dateFormat.format(offer.getExpirationDate()))
+        document.getElementById("project-title").text(offer.getProjectTitle())
+        document.getElementById("project-description").text(offer.getProjectObjective())
+        if (offer.getExperimentalDesign()) {
+            document.getElementById("experimental-design").text(offer.getExperimentalDesign())
+        }
+    }
+
+    private static void fillCostSummary(Document document, OfferContent offer) {
+        // the overhead costs
+        document.select(".overheads  .data-generation  > .costs").each {element ->
+            element.text(Currency.format(offer.getOverheadsDataGeneration()))
+        }
+        document.select(".overheads  .data-analysis  > .costs").each {element ->
+            element.text(Currency.format(offer.getOverheadsDataAnalysis()))
+        }
+        document.select(".overheads  .data-management  > .costs").each {element ->
+            element.text(Currency.format(offer.getOverheadsProjectManagementAndDataStorage()))
+        }
+        document.select(".overheads  > .totals-section  > .costs").each {element ->
+            element.text(Currency.format(offer.getOverheadTotal()))
+        }
+        // other fields
+        document.select(".total-costs .net > .costs").each {element ->
+            element.text(Currency.format(offer.getNetCost()))
+        }
+        document.select(".total-costs .discounts > .costs").each {element ->
+            element.text(Currency.format(offer.getTotalDiscountAmount() * (-1)))
+        }
+        document.select(".total-costs .overheads > .costs").each {element ->
+            element.text(Currency.format(offer.getOverheadTotal()))
+        }
+        document.select(".total-costs .vat > .costs").each {element ->
+            element.text(Currency.format(offer.getTotalVat()))
+        }
+        document.select(".total-costs > .totals-section > .costs").each {element ->
+            element.text(Currency.format(offer.getTotalCost()))
+        }
+    }
+
+    private static void fillDataGenerationItems(Document document, OfferContent offer) {
+        Element dataGenerationTable = document.getElementById("data-generation-items")
+
+        // remove table if no items are available
+        if (offer.getDataGenerationItems().isEmpty()) {
+            dataGenerationTable.remove()
+            return
+        }
+        // clear the table
+        dataGenerationTable.select("tbody > tr").each {element ->
+            element.remove()
+        }
+        // fill the table
+        int position = 1
+        Element tableBody = dataGenerationTable.selectFirst("tbody")
+        for (OfferItem offerItem in offer.getDataGenerationItems()) {
+            OfferPosition offerPosition = OfferPosition.createProductItem(position, offerItem)
```

Comment:
Why come up with a new name again. I fear we will lose track of what is meant when. Why not call it OfferItem. And in general, do we really need to create that object? 
I would like it more if you would call a mehtod on you class that obtains the information from the postition and the offerItem itself. I dont see why you would need that object. The same for the discount. It makes is more complex than it needs to be

Reply:
I disagree. The OfferPosition is the position in HTML. It does not provide all the information and there might be multiple positions (e.g. discounts, ...) for one offer item. 
The class delegates the creation of the actual HTML body for a row in the listing.

Reply:
But then its a discount item and a normal offer item. I still dont think its a good idea to introduce yet another name which just has the same information as the other DTOs. I think the delegation for the HTML creation is fine, I dont get why you would add a second step to create an object from an object which in essence has the same information